common/ledger/snapshot/file.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package snapshot
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"hash"
+	"io"
+	"os"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+)
+
+// FileCreator creates a new file for ledger snapshot. This is expected to be used by various
+// components of ledger, such as blockstorage and statedb for exporting the relevent snapshot data
+type FileCreator struct {
```

Comment:
```suggestion
type FileWriter struct {
```
`FileReader` and `FileWriter` seem to compliment each other well. I assume we have prefixed `File` so that we can introduce `snapshot.GRPCReader` and `snapshot.GRPCWriter` later. If not, we could just rename it `Reader` and `Writer`.

Reply:
`Writer` typically means that you can open anytime and start writing. A snapshot file, is created and written to just once. So, I find `FileCreator` conveying the semantics better.

Regarding `File` prefix - without this, it would sound like a `Creator` creates the whole snapshot. Whereas, `FileCreator` conveys the semantics better, as this creates a single file towards a snapshot.

Reply:
`FileCreator` sounds like an object that helps in the creation of files -- similar to [Create()](https://github.com/golang/go/blob/master/src/os/file.go#L315) in os pkg. It does not sound like an object that helps in writing content to files. What is the difference between `Reader` and `Writer` in this specific context? While the Reader is used, why not the `Writer` instead of a `Creator` which looks bit unnatural.

Reply:
Sure, I do not have a hard stand here. If you are strong on this naming, I can change this to `Writer`